rfc-161-technical-direction-in-publishing-2023.md

```diff
@@ -0,0 +1,144 @@
+
+#### Only build new apps in exceptional circumstances 
+
+We have [lots of apps](https://docs.publishing.service.gov.uk/apps.html), far more than we can realistically maintain and iterate - we don’t want to add to this (indeed, we’d like to decrease it). We aspire to have our apps meet a broad section of needs for the majority of publishers, rather than being bespoke functionality for niche scenarios. We’ve learned many times that it’s quick to create a new app, hard to justify continuously financing them and harder still, once they have some usage, to retire or consolidate them.
+
+When new user needs emerge, that prompt the consideration of a new app, we want to consider the following:
+
+- Is this need a variation on one our existing applications meet? If so, improve the existing app
+- Can most of this need be fulfilled by an industry product? If so, prefer that - we prefer building software for what is unique to us
+- Is this new software need something the GOV.UK programme will continue investing in for the foreseeable future? If not, it’s unwise to build it.
+- Is this a need for a specific department? If so, can we abstract the specific need to a common need to avoid the ownership confusion of us building a tool for one department
+
+#### Embrace existing conventions; start new conventions collaboratively
+
+GOV.UK software, like most software, is built upon layers of convention and standards. We like conventions: they reduce developer surprises and unnecessary decisions. We don’t like competing conventions - we prefer one consistent, if non-ideal, approach to solve a problem applied to an app rather than individual developers picking their individual preference each time they face a similar problem.
+
+Many GOV.UK conventions are documented. There is the [Service Manual](https://www.gov.uk/service-manual) on the high level team approach. There are organisational tech standards in the [GDS Way](https://gds-way.cloudapps.digital/). GOV.UK has [conventions for Rails applications](https://docs.publishing.service.gov.uk/manual/conventions-for-rails-applications.html) and individual repositories can have [team agreed conventions](https://github.com/alphagov/whitehall/blob/main/docs/adr/0001-add-view-components.md). 
+
+Not all conventions are documented. Some are learned from reading the code and the early decisions the developers made; others can be learned from observing team members or through code reviews. Not all conventions are in how software is written or documented. Others are in how team members interact with each other and collaborate.
+
+Starting new conventions is a team activity not an individual one - we want the results of our teams to feel collective. Please communicate liberally in starting a convention so it has the best chance of becoming established and adopted.
+
+#### Code should be correct, clear, concise and tested
```

Comment:
I agree with everything in this section, and the "Embrace existing conventions; start new conventions collaboratively" and "Optimise for maintainability, yet prioritise security" sections - but they don't feel Publishing-specific. Should they be moved to a general "code standards" section of the GOV.UK Developer Docs?

Reply:
Yeah - I think all of these could apply broader than Publishing so we've not aimed for them to be Publishing specific, just aimed to get agreement only in Publishing (which is more relevant for the product direction one). I'll plan to keep them in here - I'd not be opposed though, should people find these useful, for them to be re-purposed/refactored into GOV.UK wide principles.